providers/tests/system/google/cloud/cloud_sql/example_cloud_sql_query_iam.py

```diff
@@ -0,0 +1,491 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""
+Example Airflow DAG that performs query in a Cloud SQL instance with IAM service account authentication.
+"""
+
+from __future__ import annotations
+
+import base64
+import json
+import logging
+import os
+import random
+import string
+from copy import deepcopy
+from datetime import datetime
+from pathlib import Path
+from typing import Any, Iterable
+
+from googleapiclient import discovery
+
+from airflow import settings
+from airflow.decorators import task
+from airflow.models.connection import Connection
+from airflow.models.dag import DAG
+from airflow.providers.google.cloud.hooks.cloud_sql import CloudSQLHook
+from airflow.providers.google.cloud.hooks.secret_manager import GoogleCloudSecretManagerHook
+from airflow.providers.google.cloud.operators.cloud_sql import (
+    CloudSQLCreateInstanceDatabaseOperator,
+    CloudSQLCreateInstanceOperator,
+    CloudSQLDeleteInstanceOperator,
+    CloudSQLExecuteQueryOperator,
+)
+from airflow.settings import Session
+from airflow.utils.trigger_rule import TriggerRule
+
+from providers.tests.system.google import DEFAULT_GCP_SYSTEM_TEST_PROJECT_ID
+
+ENV_ID = os.environ.get("SYSTEM_TESTS_ENV_ID", "default")
+PROJECT_ID = os.environ.get("SYSTEM_TESTS_GCP_PROJECT") or DEFAULT_GCP_SYSTEM_TEST_PROJECT_ID
+DAG_ID = "cloudsql_query_iam"
+REGION = "us-central1"
+HOME_DIR = Path.home()
+
+COMPOSER_ENVIRONMENT = os.environ.get("COMPOSER_ENVIRONMENT", "")
+
+
+def run_in_composer():
```

Comment:
Do we really need it like a separate function and calling it each time?  I think `if COMPOSER_ENVIRONMENT: ...`  assuming definition `COMPOSER_ENVIRONMENT = os.environ.get("COMPOSER_ENVIRONMENT", "")`  it should work.

If we want it verbose in the same way I suggest defining `RUN_IN_COMPOSER=bool(COMPOSER_ENVIRONMENT)` and using it directly.